test/src/kernels/OptionallyCoupledForce2.C

```diff
@@ -0,0 +1,49 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "OptionallyCoupledForce2.h"
+
+registerMooseObject("MooseTestApp", OptionallyCoupledForce2);
+
+template <>
+InputParameters
+validParams<OptionallyCoupledForce2>()
+{
+  InputParameters params = validParams<Kernel>();
+
+  params.addCoupledVar("v", {1, 2}, "The two coupled variables which provide the force");
+
+  return params;
+}
+
+OptionallyCoupledForce2::OptionallyCoupledForce2(const InputParameters & parameters)
+  : Kernel(parameters)
+{
+  if (coupledComponents("v") != 2)
+    mooseError("Should always have two coupled components.");
+  _v_var.resize(2);
+  _v.resize(2);
+  for (unsigned int j = 0; j < 2; ++j)
+  {
+    _v_var[j] = coupled("v", j);
+    _v[j] = &coupledValue("v", j);
+  }
```

Comment:
Looking at this object more, I do not like that it is restricted to 2 values. Why would not you make it general to work with `n` values? You can get `n` by calling `coupledComponents`, resize the vectors as usual, get the references. Then, in `computeQpResidual` you do a loop over all your variables, rather than `_v[0] + v[1]`.

Then, you do not need the `mooseError` and you do not have to test the `mooseError` either. We do not really need to test the test objects :-)

Reply:
And rename the class, so it does not have 2 in it. May be something like `OptionallyVectorCoupledForce`.